src/main/java/org/broadinstitute/hellbender/engine/LocusWalker.java

```diff
@@ -0,0 +1,152 @@
+     */
+    public CountingReadFilter makeReadFilter(){
+        return new CountingReadFilter("Wellformed", new WellformedReadFilter(getHeaderForReads()))
+                .and(new CountingReadFilter("Mapped", ReadFilterLibrary.MAPPED));
+    }
+
+    /**
+     * Get the information about how to downsample the reads. By default {@link DownsamplingMethod#NONE} is returned.
+     * Subclasses should override it to provide their own downsampling methods.
+     *
+     * @return the downsampling method for the reads
+     */
+    public DownsamplingMethod getDownsamplingMethod() {
+        // TODO: change when downsampling command line options are implemented
+        return DownsamplingMethod.NONE;
+    }
+
+    /**
+     * Marked final so that tool authors don't override it. Tool authors should override onTraversalStart() instead.
+     */
+    @Override
+    protected final void onStartup() {
+        super.onStartup();
+        if ( hasIntervals() ) {
+            reads.setTraversalBounds(intervalArgumentCollection.getTraversalParameters(getHeaderForReads().getSequenceDictionary()));
+        }
+    }
+
+    /**
+     * Implementation of locus-based traversal.
+     * Subclasses can override to provide their own behavior but default implementation should be suitable for most uses.
+     *
+     * The default implementation iterates over all positions in the reference covered by reads for all samples in the read groups, using
+     * the downsampling method provided by {@link #getDownsamplingMethod()}
+     * and including deletions only if {@link #includeDeletions()} returns {@code true}.
+     */
+    @Override
+    public void traverse() {
+        final SAMFileHeader header = getHeaderForReads();
+        // get the samples from the read groups
+        final Set<String> samples = header.getReadGroups().stream()
+                                          .map(SAMReadGroupRecord::getSample)
+                                          .collect(Collectors.toSet());
+        CountingReadFilter countedFilter = disableAllReadFilters ?
+                new CountingReadFilter("Allow all", ReadFilterLibrary.ALLOW_ALL_READS ) :
+                makeReadFilter();
+        // get the LIBS
+        LocusIteratorByState libs = new LocusIteratorByState(new ReadFilteringIterator(reads.iterator(), countedFilter), getDownsamplingMethod(), includeDeletions(), KEEP_UNIQUE_READ_LIST_IN_LIBS, samples, header);
+        // prepare the iterator
+        Spliterator<AlignmentContext> iterator = (hasIntervals()) ? new IntervalOverlappingIterator<>(libs, intervalsForTraversal, header.getSequenceDictionary()).spliterator() : libs.spliterator();
+        // iterate over each alignment, and apply the function
+        StreamSupport.stream(iterator, false)
+            .forEach(alignmentContext -> {
+                        final SimpleInterval alignmentInterval = new SimpleInterval(alignmentContext.getLocation());
```

Comment:
Since `AlignmentContext` now implements `Locatable` directly, you should be able to do `new SimpleInterval(alignmentContext)` here.